[DE][9.4 & Serverless] Attach Security Detection rule as AI Agent context (#5540)
elastic/docs-content: solutions/security/ai/agent-builder/agent-builder.md

- **AI rule creation**: On the [Detection rules (SIEM)](/solutions/security/detect-and-alert/manage-detection-rules.md) page, choose **Create a rule** → **AI rule creation**. The flyout opens with an empty rule attachment and a prefilled prompt for an {{esql}} detection rule with the main rule fields. You can edit the prompt before sending the message.
- **Rule details**: Open a rule from the list, then use **Add to chat** on the rule details page.
- **Rule form (create or edit)**: While [creating](/solutions/security/detect-and-alert/using-the-rule-ui.md) or [editing](/solutions/security/detect-and-alert/manage-detection-rules.md#edit-single-rule) a rule, use **Add to chat** to send the current rule draft.
- **Alerts flyout**: Open an alert, expand the rule summary in the flyout, then use **Add to chat**.
- **Alerts table rule flyout**: From the alerts table, open the rule flyout for an alert and use **Add to chat**.

> This flow uses the [Threat Hunting agent](/explore-analyze/ai-features/agent-builder/builtin-agents-reference.md#threat-hunting-agent) with a security session context. When it responds, the rule appears in the chat as a rich attachment that shows the rule type, description, query with syntax highlighting, index patterns, tags, severity and risk score, and schedule. It draws on the attached rule to help with detection intent, query logic, MITRE ATT&CK coverage, timing and scheduling, and rule metadata quality. 

Reviewer: Threat hunting agent was removed from Kibana in https://github.com/elastic/kibana/pull/255697

Author: Thanks for flagging this @vitaliidm! I was not aware of this until today 😅 . I just confirmed with James S. that the TH agent will be unavailable in the UI for 9.4, so I'll need to rewrite this to take that into account. I'm not sure whether this is already the case for Serverless, but will check and confirm.

Author: @vitaliidm I rewrote that part of the docs [here](https://github.com/elastic/docs-content/pull/5540/changes#diff-ee560ec24eed38f2424378aff37c487d77c4267df508d384a4070ef06426d4baR42). @benironside and @leemthompo feel free to add your thoughts too, since you're more informed on what's happening on the AB side.

By default it includes the [platform core tools](/explore-analyze/ai-features/agent-builder/tools/builtin-tools-reference.md#platform-core-tools) and [security tools](/explore-analyze/ai-features/agent-builder/tools/builtin-tools-reference.md#security-tools). You can [clone the agent](/explore-analyze/ai-features/agent-builder/custom-agents.md#create-a-new-agent) to create a version with access to additional built-in or custom tools. To learn more about the available tools, refer to [](/explore-analyze/ai-features/agent-builder/tools/custom-tools.md).

## Create and refine detection rules in Agent Builder [create-and-refine-detection-rules-in-agent-builder]

```{applies_to}
stack: ga 9.4
serverless:
  security: ga
```

You can pass a detection rule into the Agent Builder chat so you can ask questions about it, get suggestions for improving rule fields, or request an appropriate investigation guide, without copying and pasting rule content between the UI and the chat. Open Agent Builder with a rule in context from any of these places:

> - **AI rule creation**: On the [Detection rules (SIEM)](/solutions/security/detect-and-alert/manage-detection-rules.md) page, choose **Create a rule** → **AI rule creation**. The flyout opens with an empty rule attachment and a prefilled prompt for an {{esql}} detection rule with the main rule fields. You can edit the prompt before sending the message.

Author: Not seeing this flow in my testing. Might need to remove.

Reviewer: it's behind Feature flag right now

Author: ```suggestion
- **AI rule creation**: On the [Detection rules (SIEM)](/solutions/security/detect-and-alert/manage-detection-rules.md) page, choose **Create a rule > AI rule creation**. The flyout opens with an empty rule attachment and a prefilled prompt for an {{esql}} detection rule with the main rule fields. You can edit the prompt before sending the message.
```

Author: Just confirmed it's showing up in serverless!

- **Rule details**: Open a rule from the list, then use **Add to chat** on the rule details page.
- **Rule form (create or edit)**: While [creating](/solutions/security/detect-and-alert/using-the-rule-ui.md) or [editing](/solutions/security/detect-and-alert/manage-detection-rules.md#edit-single-rule) a rule, use **Add to chat** to send the current rule draft.

> - **Alerts flyout**: Open an alert, expand the rule summary in the flyout, then use **Add to chat**.

Author: Unable to test this and the next list item in my current Serverless project. Need to find a prod Security project with alerts.

You can create a workflow that:

 - Runs periodically, and initiates Attack Discovery when it runs

>  - Sends any discovered attacks to Agent Builder to analyze and create a report

Reviewer 2: ```suggestion
 - Sends any discovered attacks to an AI agent to analyze and create a report
```

 - Sends that report to a third-party incident management platform and sends alerts to your team

### Example 2: Triage an alert with a workflow
You can create a workflow that:

- Triggers automatically when a rule generates an alert

> - Provides the alert data to Agent Builder with a pre-defined prompt such as `analyze this alert, check whether it's connected to existing attacks, and identify all implicated entities`

Reviewer 2: ```suggestion
- Provides the alert data to an AI agent with a pre-defined prompt such as `analyze this alert, check whether it's connected to existing attacks, and identify all implicated entities`
```

- Manually correlate new alert with its context
- Make a triage decision

> With Agent Builder, you can automate this process to speed it up and require less user input. For example, in response to the prompt `"Analyze alert abc123. What's the entity risk score for the affected host? Are there any related attack discoveries in the last 24 hours?"` Agent Builder would take the following actions (for example via **alert-analysis** and related security tools):

Reviewer 2: ```suggestion
With Agent Builder, you can automate this process to speed it up and require less user input. For example, in response to the prompt `"Analyze alert abc123. What's the entity risk score for the affected host? Are there any related attack discoveries in the last 24 hours?"` an AI agent would take the following actions:
```

same as previous + I think can lose the parenthesis here for readability
